rskj-core/src/main/java/co/rsk/pcc/blockheader/BlockHeaderContract.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2019 RSK Labs Ltd.
+ * (derived from ethereumJ library, Copyright (c) 2016 <ether.camp>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.pcc.blockheader;
+
+import co.rsk.config.RskSystemProperties;
+import co.rsk.core.RskAddress;
+import co.rsk.pcc.NativeContract;
+import co.rsk.pcc.NativeMethod;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Precompiled contract that provides access to Block Header fields (coinbase, minimum gas price, block hash, merged
+ * mining tags, bitcoin header, gas limit, gas used, RSK difficulty and coinbase for uncles).
+ *
+ * @author Diego Masini
+ */
+public class BlockHeaderContract extends NativeContract {
+    private static final short MAX_DEPTH = 4000;
```

Comment:
How did you decide to have this as the max depth? Ethereum for example has a limit of 256 (see Solidity's `blockhash(uint blockNumber)`).

I also wonder why they chose to have an absolute block number instead of a relative number (depth). Did you consider both?

Reply:
The limit is based on how many blocks the block cache has, it is 5000 actually (defined in IndexedBlockStore.java) so I will change this value from 4000 to 5000, this is also the reason why we use a relative depth and not a block number.

Reply:
The number of items in the `IndexedBlockStore` cache was just a good guess of what would make things speed up, and we didn't pay much attention because it didn't impact consensus.

I would suggest investigating why Ethereum chose 256 as their max depth, or analyze our use case in more depth.

E.g. how does it relate to the `co.rsk.config.RemascConfig#getMaturity` value?

cc @SergioDemianLerner

Reply:
The comment you added is the least we could do, thanks! (`// See: REMASC Maturity`)